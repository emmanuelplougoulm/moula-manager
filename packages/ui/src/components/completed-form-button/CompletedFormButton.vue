<template>
  <GroupButtonContainer type="completed" :disabled="disabled" @click="click">
    <template #left-icon>
      <Done class="left-icon" />
    </template>
    <template #text>
      <span class="text--highlighted text-default-bold">
        {{ title }}
      </span>
      <div v-for="(tab, index) in subtitle" :key="`subtitle-${index}`" class="subtitle-container">
        <span v-for="item in tab" :key="item" class="text--highlighted text-small-regular">
          {{ item }}
        </span>
      </div>
    </template>
    <template #right-icon>
      <Edit class="right-icon" />
    </template>
  </GroupButtonContainer>
</template>

<script setup lang="ts">
import GroupButtonContainer from '../group-button-container/GroupButtonContainer.vue';
import Edit from '../icons/Edit.vue';
import Done from '../../assets/svg/Done.svg';
import type { PropType } from 'vue';

const props = defineProps({
  title: {
    type: String,
    required: true
  },
  subtitle: {
    type: Array as PropType<string[][]>,
    default: () => [[]]
  },
  disabled: {
    type: Boolean,
    default: false
  }
});

const emit = defineEmits(['click']);

const click = () => {
  if (!props.disabled) {
    emit('click');
  }
};
</script>

<style scoped>
.text--highlighted {
  color: var(--primary-text);
}

.subtitle-container {
  display: flex;
  flex-direction: column;
}

.left-icon {
  height: 2.5rem;
  width: 2.5rem;
}

.right-icon {
  width: 1.125rem;
  height: 1.125rem;
}
</style>
