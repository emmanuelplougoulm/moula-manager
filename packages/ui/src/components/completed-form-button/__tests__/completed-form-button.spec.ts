import Icons from '../../icons/index';
import { mount } from '@vue/test-utils';
import CompletedFormButton from '../CompletedFormButton.vue';

const button = "[data-testid='button']";
const subtitle = [
  ['Phone number', 'Email address'],
  ['Address', 'Zipcode', 'City']
];

describe('CompletedFormButton tests', () => {
  it('should display the component with one subtitle', () => {
    const wrapper = mount(CompletedFormButton, {
      global: {
        plugins: [Icons]
      },
      props: {
        title: 'Title',
        subtitle: [['Phone number', 'Email address']]
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should display the component with several subtitles', () => {
    const wrapper = mount(CompletedFormButton, {
      global: {
        plugins: [Icons]
      },
      props: {
        title: 'Title',
        subtitle: subtitle
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('should emit a click event when the button is clicked', async () => {
    const wrapper = mount(CompletedFormButton, {
      props: {
        title: 'Title',
        subtitle: subtitle
      }
    });
    await wrapper.find(button).trigger('click');
    expect(wrapper.emitted()).toHaveProperty('click');
  });

  it('should not emit a click event when the button is clicked if the disabled prop is set to true', async () => {
    const wrapper = mount(CompletedFormButton, {
      props: {
        title: 'Title',
        subtitle: subtitle,
        disabled: true
      }
    });
    await wrapper.find(button).trigger('click');
    expect(wrapper.emitted()).not.toHaveProperty('click');
  });
});
