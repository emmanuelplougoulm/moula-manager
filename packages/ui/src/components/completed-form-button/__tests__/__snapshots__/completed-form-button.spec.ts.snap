// Vitest Snapshot v1

exports[`CompletedFormButton tests > should display the component with one subtitle 1`] = `
"<div data-v-d42ab394=\\"\\" data-v-c71fe769=\\"\\" class=\\"container\\"><button data-v-d42ab394=\\"\\" type=\\"button\\" class=\\"button button--completed\\" data-testid=\\"button\\">
    <div data-v-d42ab394=\\"\\" class=\\"icon-text-container\\">
      <div data-v-d42ab394=\\"\\" class=\\"icon-container\\"><svg data-v-c71fe769=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"40\\" height=\\"40\\" fill=\\"none\\" class=\\"left-icon\\">
          <path fill=\\"currentColor\\" fill-rule=\\"evenodd\\" d=\\"M14.156 30a.84.84 0 0 1-.59-.243l-9.988-9.995a.834.834 0 0 1 1.178-1.179l9.4 9.405 19.41-19.411a.836.836 0 0 1 1.18 0 .836.836 0 0 1 0 1.18l-20 20a.84.84 0 0 1-.59.243Z\\" clip-rule=\\"evenodd\\"></path>
        </svg></div>
      <div data-v-d42ab394=\\"\\" class=\\"text-container\\"><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-default-bold\\">Title</span>
        <div data-v-c71fe769=\\"\\" class=\\"subtitle-container\\"><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Phone number</span><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Email address</span></div>
      </div>
    </div>
    <div data-v-d42ab394=\\"\\" class=\\"right-icon--completed\\">
      <div data-v-b9dd0015=\\"\\" data-v-c71fe769=\\"\\" class=\\"container right-icon\\"><svg data-v-b9dd0015-s=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\" viewBox=\\"0 96 960 960\\">
          <path d=\\"M795.696 394.522 665.435 264.261l39.739-40.304q18.696-18.696 46.174-18.479 27.478.218 47.87 19.479l37.347 36.782q20.392 19.826 18.979 47.087-1.414 27.261-20.109 45.957l-39.739 39.739Zm-41.435 41.435L247.435 942.783H117.174V812.522l506.261-506.261 130.826 129.696Z\\"></path>
        </svg></div>
    </div>
  </button></div>"
`;

exports[`CompletedFormButton tests > should display the component with several subtitles 1`] = `
"<div data-v-d42ab394=\\"\\" data-v-c71fe769=\\"\\" class=\\"container\\"><button data-v-d42ab394=\\"\\" type=\\"button\\" class=\\"button button--completed\\" data-testid=\\"button\\">
    <div data-v-d42ab394=\\"\\" class=\\"icon-text-container\\">
      <div data-v-d42ab394=\\"\\" class=\\"icon-container\\"><svg data-v-c71fe769=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"40\\" height=\\"40\\" fill=\\"none\\" class=\\"left-icon\\">
          <path fill=\\"currentColor\\" fill-rule=\\"evenodd\\" d=\\"M14.156 30a.84.84 0 0 1-.59-.243l-9.988-9.995a.834.834 0 0 1 1.178-1.179l9.4 9.405 19.41-19.411a.836.836 0 0 1 1.18 0 .836.836 0 0 1 0 1.18l-20 20a.84.84 0 0 1-.59.243Z\\" clip-rule=\\"evenodd\\"></path>
        </svg></div>
      <div data-v-d42ab394=\\"\\" class=\\"text-container\\"><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-default-bold\\">Title</span>
        <div data-v-c71fe769=\\"\\" class=\\"subtitle-container\\"><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Phone number</span><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Email address</span></div>
        <div data-v-c71fe769=\\"\\" class=\\"subtitle-container\\"><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Address</span><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">Zipcode</span><span data-v-c71fe769=\\"\\" class=\\"text--highlighted text-small-regular\\">City</span></div>
      </div>
    </div>
    <div data-v-d42ab394=\\"\\" class=\\"right-icon--completed\\">
      <div data-v-b9dd0015=\\"\\" data-v-c71fe769=\\"\\" class=\\"container right-icon\\"><svg data-v-b9dd0015-s=\\"\\" xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"48\\" height=\\"48\\" viewBox=\\"0 96 960 960\\">
          <path d=\\"M795.696 394.522 665.435 264.261l39.739-40.304q18.696-18.696 46.174-18.479 27.478.218 47.87 19.479l37.347 36.782q20.392 19.826 18.979 47.087-1.414 27.261-20.109 45.957l-39.739 39.739Zm-41.435 41.435L247.435 942.783H117.174V812.522l506.261-506.261 130.826 129.696Z\\"></path>
        </svg></div>
    </div>
  </button></div>"
`;
